src/hdusd/usd_nodes/nodes/transformations.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2020 Advanced Micro Devices, Inc
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ********************************************************************
+import bpy
+import math
+
+from pxr import Usd, UsdGeom
+
+from .base_node import USDNode
+
+
+EDITABLE_TYPES = ('Xform',)
+
+
+def enabled(val):
+    if val is None:
+        return False
+
+    if isinstance(val, float) and math.isclose(val, 0.0):
+        return False
+
+    if isinstance(val, tuple) and \
+            math.isclose(val[0], 0.0) and \
+            math.isclose(val[1], 0.0) and \
+            math.isclose(val[2], 0.0):
+        return False
+
+    return True
+
+
+class TransformNode(USDNode):
+    """Transforms input data"""
+    bl_idname = 'usd.TransformNode'
+    bl_label = "Transform"
+
+    def update_data(self, context):
+        self.reset()
+
+    # region get/set
+    def set_offset_x(self, value):
+        self["offset_x"] = value
+
+    def set_offset_y(self, value):
+        self["offset_y"] = value
+
+    def set_offset_z(self, value):
+        self["offset_z"] = value
+
+    def get_offset_x(self):
+        return self.get("offset_x", 0.0)
+
+    def get_offset_y(self):
+        return self.get("offset_y", 0.0)
+
+    def get_offset_z(self):
+        return self.get("offset_z", 0.0)
+
+    def set_scale_x(self, value):
+        self["scale_x"] = value
+
+    def set_scale_y(self, value):
+        self["scale_y"] = value
+
+    def set_scale_z(self, value):
+        self["scale_z"] = value
+
+    def get_scale_x(self):
+        return self.get("scale_x", 1.0)
+
+    def get_scale_y(self):
+        return self.get("scale_y", 1.0)
+
+    def get_scale_z(self):
+        return self.get("scale_z", 1.0)
+
+    def set_rotate_x(self, value):
+        self["rotate_x"] = value
+
+    def set_rotate_y(self, value):
+        self["rotate_y"] = value
+
+    def set_rotate_z(self, value):
+        self["rotate_z"] = value
+
+    def get_rotate_x(self):
+        return self.get("rotate_x", 0.0)
+
+    def get_rotate_y(self):
+        return self.get("rotate_y", 0.0)
+
+    def get_rotate_z(self):
+        return self.get("rotate_z", 0.0)
+    # endregion
+
+    # region properties
+    name: bpy.props.StringProperty(
+        name="Xform name",
+        description="Name for USD root primitive",
+        default="Transform",
+        update=update_data
+    )
+
+    toggle_offset: bpy.props.BoolProperty(update=update_data)
+    offset_x: bpy.props.FloatProperty(
```

Comment:
1. Just use default value, no need for getters and setters.
2. Try to use FloatVectorProperty
3. Rename offset -> translation, rotate -> rotation